PVTPackage/source/MultiphaseSystem/PhaseSplitModel/BlackOilFlash.cpp

```diff
@@ -44,89 +42,95 @@ bool BlackOilFlash::computeEquilibrium( BlackOilFlashMultiphaseSystemProperties
-  const double & oilSurfaceMoleDensity = m_oilPhaseModel.getSurfaceMoleDensity();
-  const double & oilSurfaceMassDensity = m_oilPhaseModel.getSurfaceMassDensity();
-  const double rsSat = m_oilPhaseModel.computeRs( pressure );
-
-  // GAS
-  const double & gasSurfaceMoleDensity = m_gasPhaseModel.getSurfaceMoleDensity();
-  const double & gasSurfaceMassDensity = m_gasPhaseModel.getSurfaceMassDensity();
-  const double rvSat = m_gasPhaseModel.computeRv( pressure );
-
-  // Phase State - Negative flash type
-  double const Ko = rvSat * ( oilSurfaceMoleDensity + gasSurfaceMoleDensity * rsSat ) / ( gasSurfaceMoleDensity + oilSurfaceMoleDensity * rvSat );
-  double const Kg = ( oilSurfaceMoleDensity + gasSurfaceMoleDensity * rsSat ) / ( rsSat * ( gasSurfaceMoleDensity + oilSurfaceMoleDensity * rvSat ) );
-  double const V = zo / ( 1. - Kg ) + zg / ( 1. - Ko );
-
-  if( ( 0 < V ) && ( V < 1 ) )  //Two-phase or both oil and gas saturated
+  // check feed first, and if only water is present (e.g., pure water injection), do nothing
+  if( zw >= 1.0 )
   {
-    // Phase Fractions
-    sysProps.setOilFraction( 1. - V - zw );
-    sysProps.setGasFraction( V );
+    // phase fractions
+    sysProps.setOilFraction( 0.0 );
+    sysProps.setGasFraction( 0.0 );
     sysProps.setWaterFraction( zw );
-    // OIL
-    const double tmpOil = oilSurfaceMoleDensity / ( oilSurfaceMoleDensity + gasSurfaceMoleDensity * rsSat );
-    const std::vector< double > oilMoleComposition{ tmpOil, 1. - tmpOil, 0. }; // FIXME always 0.
-    sysProps.setOilMoleComposition( oilMoleComposition );
-
-    auto const oilSaturatedProperties = m_oilPhaseModel.computeSaturatedProperties( pressure, gasSurfaceMoleDensity, gasSurfaceMassDensity );
-    sysProps.setOilModelProperties( oilSaturatedProperties );
 
-    // GAS
-    const double tmpGas = gasSurfaceMoleDensity / ( gasSurfaceMoleDensity + oilSurfaceMoleDensity * rvSat );
-    const std::vector< double > gasMoleComposition{ 1. - tmpGas, tmpGas, 0. }; // FIXME always 0.
-    sysProps.setGasMoleComposition( gasMoleComposition );
-
-    auto const gasSaturatedProperties = m_gasPhaseModel.computeSaturatedProperties( pressure, oilSurfaceMoleDensity, oilSurfaceMassDensity );
-    sysProps.setGasModelProperties( gasSaturatedProperties );
-
-    // Oil ln fugacity
-    const std::vector< double > oilLnFugacity{
-      std::log( oilMoleComposition[0] * Ko * pressure ),
-      std::log( oilMoleComposition[1] * Kg * pressure ),
-      std::log( 1. )
-    };
-    sysProps.setOilLnFugacity( oilLnFugacity );
-
-    // Gas and water ln fugacity
-    const std::vector< double > gasAndWaterLnFugacity{
-      std::log( gasMoleComposition[0] * pressure ),
-      std::log( gasMoleComposition[1] * pressure ),
-      std::log( 1. ) // FIXME water fugacities are always std::log( 1. )
-    };
-    sysProps.setGasLnFugacity( gasAndWaterLnFugacity );
-    sysProps.setWaterLnFugacity( gasAndWaterLnFugacity ); // FIXME It's the same values for water and gas?
```

Comment:
Are these `setGasLnFugacity`, `setWaterLnFugacity`, `setOilLnFugacity` still use somewhere?
You may want to remove them in that case.

Reply:
Done, I removed them